### ISSUE TITLE
MaxGatedFilterNet

### ISSUE THREAD

**Issue by lihuiliullh** [HKUST-KnowComp/NeuralSubgraphCounting#2]
Hi Xin Liu:

class MaxGatedFilterNet(nn.Module):
    def __init__(self):
        super(MaxGatedFilterNet, self).__init__()
    
    def forward(self, p_x, g_x):
        max_x = torch.max(p_x, dim=1, keepdim=True)[0]
        if max_x.dim() == 2:
            return g_x <= max_x
        else:
            return (g_x <= max_x).all(keepdim=True, dim=2)

May I know the meaning of MaxGatedFilterNet? It looks different from the Filter network in the paper. Can you give an example to show how it works?

**Comment by seanliu96** [comment]
Hi,

It could make the results better.
Let's consider a graph with several nodes with label 0,1,2,3,4,5,... while the pattern has three nodes with label 0,1,2,3.
As we know, nodes with label 4,5..., 15 are not the candidate nodes hence we want to remove them. Consider the encoding method again, label 0, 1, 2, 3 will be encoded as 0101...0101, 0101...0110, 0101...1001, 0101...1010. And the max_x becomes 0101...1111. Hence we could use 0101...1111 to filter nodes with label 4,5,... because they must have at least 1 in the 010... part.
